src/blog/jwt-identification-and-hot-streams/index.md

```diff
@@ -0,0 +1,172 @@
+  <source src="./connect-benchmark.h264.mp4" ttype="video/mp4; codecs=avc1.4D401E,mp4a.40.2">
+</video>
+</figure>
+
+More than 2x faster! What's more important, we reduce the stress on the RPC server. That could be especially useful during _connection avalanches_.
+
+> JWT identification allows you to standardize the authentication flow for WebSockets, saves from cross-site WebSocket hijacking and brings a performance boost!
+
+And that's what you can get with AnyCable PRO and a little companion gem, [anycable-rails-jwt][], which adds helpers to generate tokens.
+
+The only question left is how to handle tokens expiration gracefully? AnyCable Go uses a specific disconnect reason to distinguish expiration from unauthorized access (sends a `{"type":"disconnect","reason":"token_expired"}` message). You can use it to refresh the token. We plan to add a built-in refresh functionality to our [anycable-client][] library soon. Stay tuned!
+
+## Adding some "hot wires" to the equation
+
+We've collected more than a hundred responses from Action Cable / AnyCable users around the world, and found that a good portion of them build applications on top of [Hotwire][hotwire] or [Stimulus Reflex][sr] (+ [CableReady][cr]). Which is pretty cool: [_frontendless_ Rails frontend][frontendless-rails] is getting more and more traction!
+
+What can AnyCable give to these users? Let's take a look at how Turbo Streams work with Rails.
+
+We have the [turbo-rails][] gem, which integrate Action Cable with Hotwire. All you need is to drop a helper into your template to start consuming streams:
+
+```erb
+<%= turbo_stream_from some_model %>
+```
+
+The helper adds a [`<turbo-cable-stream-source>` element][turbo-cable-stream-source], which initiates an Action Cable subscription. The corresponding server-side code is as follows:
+
+```ruby
+class Turbo::StreamsChannel < ActionCable::Channel::Base
+  extend Turbo::Streams::Broadcasts, Turbo::Streams::StreamName
+
+  def subscribed
+    if verified_stream_name = self.class.verified_stream_name(params[:signed_stream_name])
+      stream_from verified_stream_name
+    else
+      reject
+    end
+  end
+end
+```
+
+The `#verified_stream_name` method uses an [`ActiveSupport::MessageVerifier`][message-verifier] to decode and verify the stream name. The subscription is rejected if the signed stream name is invalid.
+
+The code above is structurely the same as the one we wrote for JWT identification. What does it mean? We can "move" the implementation to AnyCable Go server and **create subscriptions without touching RPC**. Again.
+
+That's exactly what we did as a part of the [Signed streams][signed-streams-docs] feature. And we did it not only for Turbo Streams, but for Cable Ready as well (since its [`#stream_from` implementation][cr-stream-from] is almost the same).
+
+> Combining JWT identification with signed streams, it is possible to avoid running RPC server at all in case you only use Hotwire/CableReady functionality.
```

Comment:
By combining JWT identification with signed streams, it's possible to completely avoid running an RPC server, in case you only need to use Hotwire/CableReady functionality.

Reply:
@palkan i think it's better to avoid "/" here. can we use "and" or "or" instead?